sdk/src/main/java/io/opentdf/platform/sdk/Autoconfigure.java

```diff
@@ -0,0 +1,749 @@
+package io.opentdf.platform.sdk;
+
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Supplier;
+
+import io.opentdf.platform.policy.KeyAccessServer;
+import io.opentdf.platform.policy.attributes.GetAttributeValuesByFqnsRequest;
+import io.opentdf.platform.policy.attributes.GetAttributeValuesByFqnsResponse;
+import io.opentdf.platform.policy.attributes.GetAttributeValuesByFqnsResponse.AttributeAndValue;
+import io.opentdf.platform.policy.Attribute;
+import io.opentdf.platform.policy.Value;
+import io.opentdf.platform.policy.AttributeValueSelector;
+import io.opentdf.platform.policy.AttributeRuleTypeEnum;
+
+// Error handling class
+class AutoConfigureException extends IOException {
+    public AutoConfigureException(String message) {
+        super(message);
+    }
+}
+
+// Attribute rule types: operators!
+class RuleTypes {
+    public static final String HIERARCHY = "hierarchy";
+    public static final String ALL_OF = "allOf";
+    public static final String ANY_OF = "anyOf";
+    public static final String UNSPECIFIED = "unspecified";
+    public static final String EMPTY_TERM = "DEFAULT";
+}
+
+
+public class Autoconfigure {
+
+    public static Logger logger = LoggerFactory.getLogger(TDF.class);
+
+    public static class SplitStep {
+        public String kas;
+        public String splitID;
+
+        public SplitStep(String kas, String splitId) {
+            this.kas = kas;
+            this.splitID = splitId;
+        }
+
+        public boolean equals(Object obj) {
```

Comment:
If you define `equals` you should generally define `hashCode` so the value objects can be used in `HashSet` and as keys